package-info.java in unit test package causing "Modules must not extend from other classes" error

My team develops our Android app on Macbooks using Android Studio with Gradle 1.7. We are currently using Dagger 1.1.0 for injecting our dependencies and we have been compiling successfully both the app as well as the tests. However when trying to compile our tests on our Jenkins server which is running Red Hat 4.1.1-52, we've seen "strange" Dagger compiler errors saying "Modules must not extend from other classes" despite our Modules not actually extending from any classes. This only repros when compiling the tests, not the app itself. I was able to reduce the repro by writing a small sample Android project that reproduces the Dagger compiler error on our Jenkins machine.

Project structure:
AndroidManifest.xml
src -> com -> sample -> (Sample.java, SampleItem.java, SampleModule.java)
tests -> com -> sample -> (SampleTest.java, package-info.java)

The SampleTest uses the SampleModule from the src tree.
When I run the Gradle build task, the app builds successfully, however when I run the compileTest task, the result is:

:SampleApp:compileTest
error: Modules must not extend from other classes: com.sample.SampleModule
1 error
:SampleApp:compileTest FAILED

FAILURE: Build failed with an exception.

When I delete the package-info.java file in the tests tree, the tests compile successfully.

Here is the Gradle version on the machine that repros the issue:

Gradle 1.7
Build time:   2013-08-06 11:19:56 UTC
Build number: none
Revision:     9a7199efaf72c620b33f9767874f0ebced135d83

Groovy:       1.8.6
Ant:          Apache Ant(TM) version 1.8.4 compiled on May 22 2012
Ivy:          2.2.0
JVM:          1.7.0_17 (Oracle Corporation 23.7-b01)
OS:           Linux 2.6.18-128.el5a02xen amd64

The zip to the Sample project that reproduces this issue can be found here: https://docs.google.com/file/d/0B5kyJ7jtiu8IQ3ZNVnpLYTNFdWs/edit?usp=sharing

It's not an actual Android project that you can run. It's simply a project to reproduce the compiler error. If you cd into SampleProject, run ./gradlew compileTest to build the tests.

--- a/compiler/src/main/java/dagger/internal/codegen/ModuleAdapterProcessor.java
+++ b/compiler/src/main/java/dagger/internal/codegen/ModuleAdapterProcessor.java
@@ -213,7 +213,7 @@ public final class ModuleAdapterProcessor extends AbstractProcessor {
       TypeElement moduleType = (TypeElement) module;
 
       // Verify that all modules do not extend from non-Object types.
-      if (!moduleType.getSuperclass().equals(objectType)) {
+      if (!types.isSameType(moduleType.getSuperclass(), objectType)) {
         error("Modules must not extend from other classes: " + elementToString(module), module);
       }
 

